### PR TITLE
fix: start native orders "samples" at 0 & don't skip neg adj orders

### PR DESCRIPTION
**Pull request by kimpers**
## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-swapper/src/utils/market_operation_utils/fills.ts [hunk 1]
```diff
@@ -126,8 +127,8 @@ export function nativeOrdersToFills(
             side === MarketOperation.Sell ? clippedOutput.minus(outputPenalty) : clippedOutput.plus(outputPenalty);
         const adjustedRate =
             side === MarketOperation.Sell ? adjustedOutput.div(clippedInput) : clippedInput.div(adjustedOutput);
-        // Skip orders with rates that are <= 0.
-        if (adjustedRate.lte(0)) {
+        // Optionally skip orders with rates that are <= 0.
+        if (filterNegativeAdjustedRateOrders && adjustedRate.lte(0)) {
```
Comment (kimpers) [inline packages/asset-swapper/src/utils/market_operation_utils/fills.ts:131]: If the rust router chose a native order we want to use it, even if it has negative adjusted rate. In this case it was the best option from all the bad options 🙈

Comment (mzhu25) [reply packages/asset-swapper/src/utils/market_operation_utils/fills.ts:131]: do we ever _need_ to skip over orders with rates <= 0?

Comment (mzhu25) [reply packages/asset-swapper/src/utils/market_operation_utils/fills.ts:131]: is the assumption here that skipping them saves us some compute time?

Comment (mzhu25) [reply packages/asset-swapper/src/utils/market_operation_utils/fills.ts:131]: 😫 pain

**Review by mzhu25 — APPROVED** [review]

**Pull request merged**